tests/trainer/logging_tests/test_eval_loop_logging_1_0.py

```diff
@@ -419,3 +418,366 @@ def test_dataloader(self):
     assert len(results) == 1
     # error : It is wrong there. `y` should equal test_loss_epoch
     assert results[0]['test_loss'] == results[0]['y']
+
+
+def test_log_works_in_val_callback(tmpdir):
+    """
+    Tests that log can be called within callback
+    """
+    os.environ['PL_DEV_DEBUG'] = '1'
+
+    class TestCallback(callbacks.Callback):
+
+        # helpers
+        count = 1
+        choices = [False, True]
+        # used to compute expected values
+        callback_funcs_called = collections.defaultdict(list)
+        funcs_called_count = collections.defaultdict(int)
+        funcs_attr = {}
+
+        def make_logging(self, pl_module: pl.LightningModule, func_name,
```

Comment:
don't we have this already in another test? #4495

Reply:
Similar idea, but specific to test and multi-dataloaders.